oracle/pkg/l1Listener/l1Listener.go

```diff
@@ -159,29 +179,53 @@ func (l *L1Listener) watchL1Block(ctx context.Context) error {
 				l.logger.Error("failed to get block number", "error", err)
 				continue
 			}
+			// We need to get the previous block number because the current block has finalized header
+			blockNum = blockNum - 1
 
 			if blockNum <= uint64(currentBlockNo) {
 				continue
 			}
 
 			for b := uint64(currentBlockNo) + 1; b <= blockNum; b++ {
-				header, err := l.l1Client.HeaderByNumber(ctx, big.NewInt(int64(b)))
+				header, err := l.l1Client.HeaderByNumber(ctx, new(big.Int).SetUint64(b+1))
 				if err != nil {
 					l.logger.Error("failed to get header", "block", b, "error", err)
 					continue
 				}
 
-				winner := string(bytes.ToValidUTF8(header.Extra, []byte("�")))
+				// End of changes needed to be done.
+				var builderPubKey string
+				l.logger.Info("querying relay", "block", b, "hash", header.ParentHash.Hex())
+				for i := 0; i < 4; i++ {
+					builderPubKey, err = l.relayQuerier.Query(int64(b), header.ParentHash.Hex())
+					if err == nil {
+						break
+					}
+					l.logger.Info("block not found in relay, retrying", "block", b, "error", err, "attempt", i+1)
+					time.Sleep(500 * time.Millisecond)
+				}
+
+				if err != nil {
+					l.logger.Info("block not found in relay", "block", b, "error", err)
```

Comment:
In case of a error, is log supposed to be an info?

Reply:
This is still a happy path operation, see my comment above.